src/components/TodoFilter/TodoFilter.tsx

```diff
@@ -1,30 +1,70 @@
-export const TodoFilter = () => (
-  <form className="field has-addons">
-    <p className="control">
-      <span className="select">
-        <select data-cy="statusSelect">
-          <option value="all">All</option>
-          <option value="active">Active</option>
-          <option value="completed">Completed</option>
-        </select>
-      </span>
-    </p>
-
-    <p className="control is-expanded has-icons-left has-icons-right">
-      <input
-        data-cy="searchInput"
-        type="text"
-        className="input"
-        placeholder="Search..."
-      />
-      <span className="icon is-left">
-        <i className="fas fa-magnifying-glass" />
-      </span>
-
-      <span className="icon is-right" style={{ pointerEvents: 'all' }}>
-        {/* eslint-disable-next-line jsx-a11y/control-has-associated-label */}
-        <button data-cy="clearSearchButton" type="button" className="delete" />
-      </span>
-    </p>
-  </form>
-);
+import React, { useState } from 'react';
+import { IQuery, SortField } from '../../types/Filter';
+
+interface ITodoFilter {
+  setQuery: React.Dispatch<React.SetStateAction<IQuery>>;
+}
+
+export const TodoFilter: React.FC<ITodoFilter> = ({ setQuery }) => {
+  const [inputValue, setInputValue] = useState('');
+
+  const handleSelect = (event: React.ChangeEvent<HTMLSelectElement>) => {
+    const status = event.target.value as SortField;
+
+    setQuery(prev => {
+      return { ...prev, status };
+    });
+  };
+
+  const handleInput = (event: React.ChangeEvent<HTMLInputElement>) => {
+    setInputValue(event.target.value);
+
+    setQuery(prevQuery => {
+      return { ...prevQuery, query: event.target.value };
+    });
+  };
+
+  const handleDeleteButton = () => {
+    setQuery({ status: 'all', query: '' });
+    setInputValue('');
+  };
+
+  return (
+    <form className="field has-addons">
+      <p className="control">
+        <span className="select">
+          <select onChange={handleSelect} data-cy="statusSelect">
+            <option value="all">All</option>
+            <option value="active">Active</option>
+            <option value="completed">Completed</option>
+          </select>
```

Comment:
This select goes in uncontrolled mode, while the input is used in controlled mode

It is not critical, but not perfect as well - if you pass query as prop, you can make it consistent